source/modules.tex

```diff
@@ -554,8 +554,7 @@
 
 \pnum
 A translation unit has an \defn{interface dependency} on a translation unit \tcode{U}
-if it contains a \grammarterm{module-declaration} or
-\grammarterm{module-import-declaration} that imports \tcode{U} or if it has
+if it imports \tcode{U} or if it has
```

Comment:
I don't think we define what it means for a translation unit to import a translation unit, do we?

Reply:
[basic.scope.namespace]/2 and the note in [module.import]/2 seem to think it's meaningful, and I took (say) "a translation unit defines" as being the natural converse of common phrases like "defined in a translation unit".  If you prefer, I'll add at least a parenthetical definition to keep this phrasing in general&mdash;it gave me the strong impression that there was some other way to import a translation unit that didn't constitute an interface dependency.

Reply:
I think [basic.scope.namespace]/2 could do with a little cleanup:

> If a translation unit *Q* is imported <del>into</del> <ins>by a declaration in</ins> a translation unit *R*, ...

but the rest of the sentence already makes it contextually clear what it's referring to ("the first *module-import-declaration* or *module-declaration* in *R* that imports *Q*").

[module.import]/2's note is imprecise, but it's non-normative and seems clear enough in that context ("the importing translation unit after the import declaration" suggests that "the importing translation unit" is the one that contains the import declaration). I would prefer that we keep this normative wording precise.

I'd be OK with adding a definition for what it means for a translation unit to import another translation unit and taking this change. I'm not sure we'd use the definition very much, since we typically do care about the point of import.

Reply:
How about

> A translation unit has an _interface dependency_ on a translation unit _U_ if it contains a <ins>declaration (possibly a</ins> *module-declaration*<del> or *module-import-declaration*</del><ins>)</ins> that imports _U_ or if &hellip;

?  I just want to avoid the suggestion that there are other ways of importing that don't count for this.

Reply:
Sounds fine to me.

Reply:
Done.

Reply:
I finally found the other place I remembered seeing this: [module.reach]/1, which _does_ use it with a point of import.